pkg/control/pubcontrol/pub_control.go

```diff
@@ -56,19 +64,203 @@ func (c *commonControl) IsPodReady(pod *corev1.Pod) bool {
 }
 
 func (c *commonControl) IsPodUnavailableChanged(oldPod, newPod *corev1.Pod) bool {
-	// If pod.spec changed, pod will be in unavailable condition
+	// If pod.spec changed, pod may be in unavailable condition
 	if !reflect.DeepEqual(oldPod.Spec, newPod.Spec) {
-		klog.V(3).InfoS("Pod specification changed, and maybe cause unavailability", "pod", klog.KObj(newPod))
-		return true
+		if c.canResizeInplace(oldPod, newPod) {
+			klog.V(3).InfoS("Pod resize inplace", "pod", klog.KObj(newPod))
+			// do not return any value, continue to check unavailable label
+		} else {
+			klog.V(3).InfoS("Pod specification changed, and maybe cause unavailability", "pod", klog.KObj(newPod))
+			return true
+		}
 	}
 	// pod add unavailable label
 	if !appspub.HasUnavailableLabel(oldPod.Labels) && appspub.HasUnavailableLabel(newPod.Labels) {
+		klog.V(3).InfoS("Pod add unavailable label, and maybe cause unavailability", "pod", klog.KObj(newPod))
 		return true
 	}
 	// pod other changes will not cause unavailability situation, then return false
+
+	klog.V(3).InfoS("Pod other changes, and maybe not cause unavailability", "pod", klog.KObj(newPod))
 	return false
 }
 
+func (c *commonControl) isNeedProtectResizeAction(newPod *corev1.Pod) bool {
+	pub, err := c.GetPubForPod(newPod)
+	if err != nil || pub == nil {
+		return true
+	}
+
+	if isNeedPubProtection(pub, policyv1alpha1.PubResizeOperation) {
+		return true
+	}
+
+	// if protect UPDATE but featureGate InPlacePodVerticalScaling is disabled, then need to protect RESIZE
+	if isNeedPubProtection(pub, policyv1alpha1.PubUpdateOperation) &&
+		!feature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		return true
+	}
+
+	return false
+}
+
+// return true if this action won't cause unavailability
+func (c *commonControl) canResizeInplace(oldPod, newPod *corev1.Pod) bool {
+	if c.isNeedProtectResizeAction(newPod) {
```

Comment:
maybe we should not call isNeedProtectResizeAction at all, since isNeedPubProtection will be called in PodUnavailableBudgetValidatePod.